File: py/2.py
#!/usr/bin/python


def fibo(terms, limit):
    length = len(terms)
    s = terms[length - 1] + terms[length - 2]
    if s > limit:
        return terms

    terms.append(s)
    return fibo(terms, limit)

if __name__ == '__main__':
    print "The sum of the even terms of a fibo sequence under 4M is %s " % sum(filter(lambda x: not x % 2, fibo([1, 2], 4000000)))

File: go/2.go
package main

import "fmt"

func fibo(terms []int, limit int) []int {
	length := len(terms)
	r := terms[length-1] + terms[length-2]
	if r > limit {
		return terms
	}
	return fibo(append(terms, r), limit)
}

func main() {
	sum := 0
	for _, value := range fibo([]int{1, 2}, 4000000) {
		if value%2 == 0 {
			sum += value
		}
	}
	fmt.Println("The sum of the even terms of a fibo sequence under 4M is", sum)
}
